flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/aggregate/window/processors/SliceSharedWindowAggProcessor.java

```diff
@@ -123,6 +124,19 @@ public void merge(@Nullable Long mergeResult, Iterable<Long> toBeMerged) throws
         }
     }
 
+    protected long sliceStateMergeTarget(long sliceToMerge) throws Exception {
+        mergeTargetGetter.mergeTarget = null;
+        sliceSharedAssigner.mergeSlices(sliceToMerge, mergeTargetGetter);
```

Comment:
I got the point, but looks this doesn't very clean. Could we  rename `SliceMergeTargetGetter`  to `SliceMergeTargetHelper`  because it didn't expose `getter` method, and add `SetMergeTarget` `GetMergeTarget`.